html/semantics/interactive-elements/the-dialog-element/top-layer-position-relative.html

```diff
@@ -0,0 +1,54 @@
+<!DOCTYPE html>
+<link rel=author href="mailto:jarhar@chromium.org">
+<link rel=author href="mailto:falken@chromium.org">
+<link rel=help href="https://html.spec.whatwg.org/multipage/interactive-elements.html#the-dialog-element">
+<link rel=help href="https://bugs.webkit.org/show_bug.cgi?id=106538">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+
+<style>
+.green-box {
+  background-color: green;
+  height: 50px;
+  width: 50px;
+  border: none;
+  padding: 0;
+  margin: 0;
+}
+
+#left-div {
+  position: absolute;
+  top: 100px;
+  left: 100px;
+}
+
+#middle-dialog {
+  position: relative;
+  top: 100px;
+  left: 150px;
+}
+
+#right-dialog {
+  position: relative;
+  top: 100px;
+  left: 200px;
+}
+</style>
+
+<script src="/resources/check-layout.js"></script>
+
+<p>Bug <a href="http://webkit.org/b/106538">106538</a>: Top layer fails for inline elements
+<p>This tests that position 'relative' computes to 'absolute' in the top layer. The test passes if you see a single green box.</p>
+<div id="left-div" class="green-box"></div>
+<dialog id="middle-dialog" class="green-box" data-offset-x="150" data-offset-y="100"></dialog>
+<dialog id="right-dialog" class="green-box" data-offset-x="200" data-offset-y="100"></dialog>
+
+<script>
+test(() => {
+  document.getElementById('middle-dialog').showModal();
+  var rightDialog = document.getElementById('right-dialog');
+  rightDialog.showModal();
+  rightDialog.style.position = 'absolute';
+  assert_true(checkLayout('dialog'));
+}, `Position 'relative' computes to 'absolute' in the top layer.`);
+</script>
```

Comment:
@josepharhar Might be more simple (and would avoid the need for check-layout.js) if you made this a reftest similar to [modal-dialog-display-contents.html](https://github.com/web-platform-tests/wpt/blob/85aabd31cf317b21e5932f74f1b93277f3c39bb5/html/semantics/interactive-elements/the-dialog-element/modal-dialog-display-contents.html).

Reply:
Thanks for the suggestion, I did it